staging/src/kubevirt.io/api/core/v1/schema.go

```diff
@@ -941,13 +948,31 @@ type PersistentVolumeClaimVolumeSource struct {
 	Hotpluggable bool `json:"hotpluggable,omitempty"`
 }
 
+// MemoryDumpVolumeSource represents a memory dump volume source.
+// Deprecated: Use ScratchVolumeSource with Type "memoryDump" instead. This type will be removed in a future version.
 type MemoryDumpVolumeSource struct {
 	// PersistentVolumeClaimVolumeSource represents a reference to a PersistentVolumeClaim in the same namespace.
 	// Directly attached to the virt launcher
 	// +optional
 	PersistentVolumeClaimVolumeSource `json:",inline"`
 }
 
+type ScratchVolumeType string
+
+const (
+	MemoryDumpType ScratchVolumeType = "memoryDump"
+)
+
+type ScratchVolumeSource struct {
+	// PersistentVolumeClaimVolumeSource represents a reference to a PersistentVolumeClaim in the same namespace.
+	// Directly attached to the virt launcher
+	// +optional
+	PersistentVolumeClaimVolumeSource `json:",inline"`
+	// Type of the scratch volume.
+	// +optional
+	Type ScratchVolumeType `json:"type,omitempty"`
```

Comment:
Can we get rid of this field? If not, I think it would be better to have more "root" level volume types than one type with subtypes